### PR TITLE
Fix unsafe wallet function implementation

### PR DESCRIPTION
**Pull request by adamczykm**
Came across something that might cause problems soon.

### CODE REVIEW

File: src/Wallet.js [hunk 1]
```diff
@@ -8,19 +8,23 @@ exports._getNamiAddress = (nami) => () =>
...
 
 // _getNamiCollateral
-//   :: NamiConnection
+//   :: MaybeFfiHelper
+//   -> NamiConnection
 //   -> Effect (Promise String)
-exports._getNamiCollateral = (nami) => () =>
-  nami.experimental.getCollateral().then((utxos) => utxos[0]);
+exports._getNamiCollateral = maybe => (nami) => () =>
+nami.experimental.getCollateral().then((utxos) => {
```
Comment (ngua) [inline src/Wallet.js:15]: Just a formatting thing: could you indent this please?

**Review by ngua — COMMENTED** [review]
Looks fine to me. Nami "should" always return an array containing a single collateral UTxO when `getCollateral` is called, but it's probably a better idea to check this

**Review by ngua — APPROVED** [review]
Meant to approve this with my previous review

**Comment by adamczykm** [comment]
> Looks fine to me. Nami "should" always return an array containing a single collateral UTxO when `getCollateral` is called, but it's probably a better idea to check this

It doesn't. Yesterday I tried to run pk2pk example and it was returning empty array.

File: src/Wallet.js [hunk 2]
```diff
@@ -8,19 +8,23 @@ exports._getNamiAddress = (nami) => () =>
...
 exports._signTxNami = (txHex) => (nami) => () => nami.signTx(txHex);
 
 // _submitTxNami :: String -> NamiConnection -> Effect (Promise String)
 exports._submitTxNami = (txHex) => (nami) => () => {
   try {
-    nami.submitTx(txHex);
+    return nami.submitTx(txHex);
   } catch (e) {
-    console.log(e);
+      console.log(e);
```
Comment (klntsky) [inline src/Wallet.js:27]: 2 or 4 spaces? Otherwise OK.

**Review by klntsky — APPROVED** [review]
All fine

**Comment by ngua** [comment]
@adamczykm Did you set collateral in Nami?

**Pull request merged**